### ISSUE TITLE
Would it be possible to associate The Focus Accessible Buffer command with the shift+alt+f2 key combination?

### ISSUE THREAD

**Issue by jvesouza** [microsoft/vscode#192090]
Type: <b>Feature Request</b>
@meganrogge 

If I understand correctly, when I'm in the terminal and I want to access the accessible buffer, I need to press the key combination alt+f2. The problem is that in my environment this key combination is used by gnome to activate the command execution dialog, which does not allow me to use this combination to access the accessible buffer.

What I did to remedy this situation was to associate the command to the shift+alt+f2 key combination, but I had to do this for the different profiles I have.
For me it would be great if it was also possible to associate the command with the combination alt+shift+f2.

VS Code version: Code - Insiders 1.82.0-insider (f1302be1e67e3af5fbeb8bbb2ea784de7bc96150, 2023-09-01T11:05:38.381Z)
OS version: Linux x64 6.4.13-200.fc38.x86_64
Modes:

**Comment by meganrogge** [comment]
Yes, this is a good idea for the linux keybinding.

**Comment by jvesouza** [comment]
It is not working for me.
I open a terminal and press shift+alt+f2 but nothing happens.
But if I use the command "Terminal: Focus Accessible Buffer" the accessible buffer is displayed normally.
Am I doing something wrong?
Thanks.

**Comment by jvesouza** [comment]
It also seems that I am not able to access the accessible view by pressing alt+shift+f2. I press ctrl+alt+i to activate the chat, type a question, press enter and after listening to the answer, press alt+shift+f2. Focus does not move to the accessible view

**Comment by meganrogge** [comment]
I don't have linux, so cannot investigate. I suspect the keybinding is not taking effect for some reason. Can you share the output when you try this after running the command Toggle keyboard shortcuts troubleshooting pls?

**Comment by jvesouza** [comment]
Forgive my ignorance, but how can I share the output? Generating something like a gif?
Is there any accessible way to do this? If it doesn't exist, I'll need to ask my wife for help. Thank God we're not fighting!

**Comment by meganrogge** [comment]
The output of that command will be in the output channel, so you can focus the output to copy the logs.

**Comment by jvesouza** [comment]
Surely I must be doing something wrong because I didn't find anything in the log.
I performed the following steps:

1. Open a terminal in VSCode.
2. typed some commands.
3. I pressed alt+shift+f2.
4. I pressed ctrl+shift+p, selected Output: Show Output Channels and pressed enter.
5. Using the arrow keys I selected terminal and pressed enter.

Apparently no content was displayed.

**Comment by meganrogge** [comment]
Sorry, you have missed one step, which should be done before your above steps:

`after running the command Toggle keyboard shortcuts troubleshooting`

**Comment by jvesouza** [comment]
It seems like I finally understand what's going on.

The first problem is that except for misunderstandings, the shift+alt+f2 combination is probably being used by Gnome or some other application. What leads me to think this is that after I press this key combination with the active terminal and inspect the contents of the output channel, no reference is found for this key combination.

So I decided to try the key combination shift+alt+f3, configured as below:
```
{ "key": "shift+alt+f3",          "command": "workbench.action.terminal.focusAccessibleBuffer",
    "when": "accessibilityModeEnabled && terminalFocus && terminalHasBeenCreated && terminalTabFocusMode || accessibilityModeEnabled && terminalFocus && terminalHasBeenCreated && !terminalAccessibleBufferFocus || accessibilityModeEnabled && terminalFocus && terminalProcessSupported && terminalTabFocusMode || accessibilityModeEnabled && terminalFocus && terminalProcessSupported && !terminalAccessibleBufferFocus" },
    { "key": "shift+alt+f3",          "command": "editor.action.accessibleView" },
```

This configuration also did not work, but unlike the previous combination, references for the combination were displayed in the output channel, as shown below:

```
2023-09-11 05:40:13.962 [info] [KeybindingService]: | Resolving shift+alt+[F3]
2023-09-11 05:40:13.963 [info] [KeybindingService]: \ From 4 keybinding entries, matched editor.action.accessibleView, when: no when condition, source: user.
2023-09-11 05:40:14.208 [info] [KeybindingService]: / Soft dispatching keyboard event
2023-09-11 05:40:14.208 [info] [KeybindingService]: | Resolving shift+alt+[F3]
2023-09-11 05:40:14.209 [info] [KeybindingService]: \ From 4 keybinding entries, matched editor.action.accessibleView, when: no when condition, source: user.
2023-09-11 05:40:15.025 [info] [KeybindingService]: / Soft dispatching keyboard event
2023-09-11 05:40:15.026 [info] [KeybindingService]: \ Keyboard event cannot be dispatched
2023-09-11 05:40:15.029 [info] [KeybindingService]: / Soft dispatching keyboard event
2023-09-11 05:40:15.029 [info] [KeybindingService]: \ Keyboard event cannot be dispatched
2023-09-11 05:40:15.030 [info] [KeybindingService]: + Ignoring single modifier shift due to it being pressed together with other keys.
2023-09-11 05:40:16.747 [info] [KeybindingService]: / Soft dispatching keyboard event
2023-09-11 05:40:16.749 [info] [KeybindingService]: \ Keyboard event cannot be dispatched
2023-09-11 05:40:16.750 [info] [KeybindingService]: / Received  keydown event - modifiers: [ctrl], code: ControlLeft, keyCode: 17, key: Control
2023-09-11 05:40:16.751 [info] [KeybindingService]: | Converted keydown event - modifiers: [ctrl], code: ControlLeft, keyCode: 5 ('Ctrl')
2023-09-11 05:40:16.753 [info] [KeybindingService]: \ Keyboard event cannot be dispatched in keydown phase.
2023-09-11 05:40:16.771 [info] [KeybindingService]: / Soft dispatching keyboard event
2023-09-11 05:40:16.772 [info] [KeybindingService]: \ Keyboard event cannot be dispatched
2023-09-11 05:40:16.773 [info] [KeybindingService]: / Received  keydown event - modifiers: [ctrl,shift], code: ShiftLeft, keyCode: 16, key: Shift
2023-09-11 05:40:16.774 [info] [KeybindingService]: | Converted keydown event - modifiers: [ctrl,shift], code: ShiftLeft, keyCode: 4 ('Shift')
2023-09-11 05:40:16.774 [info] [KeybindingService]: \ Keyboard event cannot be dispatched in keydown phase.
2023-09-11 05:40:17.689 [info] [KeybindingService]: / Soft dispatching keyboard event
2023-09-11 05:40:17.691 [info] [KeybindingService]: | Resolving ctrl+shift+[KeyP]
2023-09-11 05:40:17.692 [info] [KeybindingService]: \ From 2 keybinding entries, matched 
```

Although I was definitely focused on the terminal, it appears that VSCode tried to execute the "editor.action.accessibleView" command.

The following configuration worked for me:

```
    { "key": "shift+alt+f3",          "command": "workbench.action.terminal.focusAccessibleBuffer",
    "when": "accessibilityModeEnabled && terminalFocus && terminalHasBeenCreated && terminalTabFocusMode || accessibilityModeEnabled && terminalFocus && terminalHasBeenCreated && !terminalAccessibleBufferFocus || accessibilityModeEnabled && terminalFocus && terminalProcessSupported && terminalTabFocusMode || accessibilityModeEnabled && terminalFocus && terminalProcessSupported && !terminalAccessibleBufferFocus" },
    { "key": "shift+ctrl+f3",          "command": "editor.action.accessibleView" },
```

In short: It seems that I cannot use the f2 key in my environment and I also cannot use the same key combination for both commands.

**Comment by meganrogge** [comment]
Thanks for debugging. I wish I could test this on linux and come up with a keybinding that would work. There shouldn't be an issue with assigning the same keybinding to both of those.

**Comment by meganrogge** [comment]
It occurred to me that you can use `ctrl+upArrow` to access this. does that work for you?

**Comment by jvesouza** [comment]
I also can't access the accessible buffer using ctrl+up. There is a bug open in Orca regarding this.

**Comment by meganrogge** [comment]
Has any keybinding you've tried worked?

**Comment by jvesouza** [comment]
Oops! It seems I made a mistake. Yes, it works. If I press ctrl+up when I'm in the terminal the focus is moved to the accessible buffer.
To be honest, I didn't know this feature existed and I found it much more practical. What doesn't work is pressing ctrl+up when I'm in the accessible buffer.

**Comment by meganrogge** [comment]
`Ctrl+up` should do nothing when you're in the accessible buffer. `Ctrl+down` will bring you back to the terminal from there.
